esmvaltool/diag_scripts/weighting/climwip/core_functions.py

```diff
@@ -41,6 +41,7 @@ def distance_matrix(values: 'np.ndarray',
 
     # pdist does not work with NaN
     not_nan = np.where(np.all(np.isfinite(values), axis=0))[0]
+    assert len(not_nan) > 0, 'All grid cells are NaN in at least one model!'
```

Comment:
it's not good practice to plop `assert` statements if not in a test module, I'd go with a check that then raises an exception if the conditional fails eg:
```python
if not not_nan:
    raise ValueError("All grid cells are NaN in at least one model!")  # maybe print out model name or such
```

Reply:
Ah fair point will do! The model name we don't have at this point. This should actually never happen. There was a bug recently where it did anyway and the error message was really confusing therefore I added that.